sdk/storage/azblob/client_test.go

```diff
@@ -606,7 +606,13 @@ func performUploadAndDownloadBufferTest(t *testing.T, _require *require.Assertio
 		if downloadCount == 0 {
 			_require.EqualValues(destBuffer, bytesToUpload[downloadOffset:])
 		} else {
-			_require.EqualValues(destBuffer, bytesToUpload[downloadOffset:downloadOffset+downloadCount])
+			_require.EqualValues(destBuffer[downloadOffset:downloadOffset+int(n)], bytesToUpload[downloadOffset:downloadOffset+int(n)])
+			if downloadCount > blobSize {
+				_require.EqualValues(blobSize, n)
```

Comment:
This can be incorrect when,
blobSize = 10, downloadOffset = 5, downloadCount = 100
n will be 5 here (download from offset 5 to 10)

Reply:
It should be `_require.EqualValues(blobSize-downloadOffset, n)`